src/services/state-persistor/state-persistor-theme.service.ts

```diff
@@ -0,0 +1,50 @@
+import { watch, watchEffect, WatchStopHandle } from 'vue'
+
+import { useThemeStore } from '@/stores/config.store'
+
+import { SP_KEY_THEME } from './state-persistor.model'
+import { storageHelper } from './storage/storage.helper'
+import { storageThemeMapper } from './state-persistor-theme.mapper'
+
+class StatePersistorThemeService {
+  bootstrap() {
+    this.restore()
+    let stop: WatchStopHandle
+    // eslint-disable-next-line prefer-const
+    stop = watchEffect(() => {
+      this.persist()
+      stop && stop() // test if exists, for HMR support
```

Comment:
Wondering why you can call `stop()` here. Doesn't it need to be `stop.stop()` ? (A little pity we need specific code for HMR support here, but I guess there is no alternative)

Reply:
`function watchEffect(effect: WatchEffect, options?: WatchOptionsBase): WatchStopHandle`
`WatchStopHandle` that is a function. Did you find an example where stop.stop() is called? because I don't understand how this could be possible.

About the HMR support: totally agree with you! Sadly I didn't find a lot of documentation about this specific point.

Reply:
Of what I understood, `watchEffect` returns a `WatchStopHandle`, which has a `stop()` method, but I couldn't find the official doc on this.

About the HMR, I'm wondering if we should treat this in the code or if we shouldn't just accept that there might be some cases where we have to reload the app during dev.

Reply:
`return unwatch` see: https://github.com/vuejs/core/blob/b49b9eff20e7b77896d6a1cc811eb5e2e5fea8f4/packages/runtime-core/src/apiWatch.ts#L390

About the HMR support here: the only part that is linked to HMR is the testing condition `stop &&`, really not a big deal for production.